### PR TITLE
pkg: user error when non-existent repo specified

### PR DESCRIPTION
**Pull request by gridbugs**
Previously it would be a code error if a user specified an invalid git repo url when pinning a package. This change makes it a user error instead.

### CODE REVIEW

File: boot/libs.ml [hunk 1]
```diff
@@ -1,3 +1,5 @@
+let executables = [ "main" ]
+
```
Comment (Leonidas-from-XIV) [inline boot/libs.ml:2]: This change looks to be unrelated?

File: src/dune_pkg/rev_store.ml [hunk 2]
```diff
@@ -157,28 +183,29 @@ let run_with_exit_code { dir; _ } ~allow_codes ~display args =
...
-  then exit_code
-  else git_code_error ~dir ~args ~exit_code ~output:[]
+  then Ok exit_code
+  else Error { Git_error.dir; args; exit_code; output = [] }
 ;;
 
 let run t ~display args =
-  run_with_exit_code t ~allow_codes:(Int.equal 0) ~display args >>| ignore
+  run_with_exit_code t ~allow_codes:(Int.equal 0) ~display args
+  >>| Result.map ~f:(fun (_ : int) -> ())
```
Comment (Leonidas-from-XIV) [inline src/dune_pkg/rev_store.ml:192]: I know there is some use in annotating the expected type to be ignored but it's also adding a lot of noise, so maybe

```suggestion
  >>| Result.map ~f:ignore
```

File: src/dune_pkg/rev_store.ml [hunk 3]
```diff
@@ -724,12 +758,32 @@ let remote =
...
         Fiber_lazy.create (fun () ->
-          let+ hits = run_capture_lines t ~display:!Dune_engine.Clflags.display command in
+          let+ hits =
+            run_capture_lines t ~display:!Dune_engine.Clflags.display command
+            >>| function
+            | Ok lines -> lines
+            | Error git_error ->
+              (match git_error.exit_code with
+               | 128 ->
+                 let command_string = String.concat ~sep:" " command in
```
Comment (Leonidas-from-XIV) [inline src/dune_pkg/rev_store.ml:773]: This will generate an invalid command if any of the items in `command` have a space (which I guess a URL could have, unless it is URL-encoded). Maybe better always quote the second item to be on the safe side.

**Review by Leonidas-from-XIV — COMMENTED** [review]
It's a useful change, my main suggestion is to have the URL and the Loc tied closer together.

File: boot/libs.ml (hunk 1 above)
Comment (Leonidas-from-XIV) [reply boot/libs.ml:2]: Yes, I know the problem :-) Usually I can get this fixed when I rebootstrap from `main` with `make bootstrap`.

File: src/dune_pkg/fetch.mli [hunk 4]
```diff
@@ -17,10 +17,12 @@ val fetch
   -> checksum:Checksum.t option
   -> target:Path.t
   -> OpamUrl.t
+  -> url_loc:Loc.t
```
Comment (rgrinberg) [inline src/dune_pkg/fetch.mli:20]: When it's not clear what which argument the location pertains to, dune's convention is usually to tuple it. So in this case, it should be `(Loc.t * OpamUrl.t)`.

File: src/dune_pkg/rev_store.ml (hunk 2 above)
Comment (rgrinberg) [reply src/dune_pkg/rev_store.ml:192]: It's pretty useful to know what's being ignored. I think the noise makes it worth it here. You could also do `Result.map ~f:(ignore : int -> unit)` I suppose

Comment (gridbugs) [reply src/dune_pkg/rev_store.ml:192]: Updated to use the `ignore` function as I think it's clearer.

**Review by rgrinberg — APPROVED** [review]

**Review by Leonidas-from-XIV — APPROVED** [review]
